### PR TITLE
fix(Tree): Using anchor for tree title

### PR DESCRIPTION
**Pull request by priyankar205**

### CODE REVIEW

File: packages/react/src/components/Tree/TreeTitle.tsx [hunk 1]
```diff
@@ -57,15 +57,13 @@ class TreeTitle extends UIComponent<ReactProps<TreeTitleProps>> {
...
+    href: '#',
     accessibility: treeTitleBehavior,
   }
 
   protected actionHandlers: AccessibilityActionHandlers = {
-    performClick: e => {
-      e.preventDefault()
-      this.handleClick(e)
-    },
+    performClick: e => this.handleClick(e),
```
Comment (layershifter) [inline packages/react/src/components/Tree/TreeTitle.tsx]: @jurokapsiar Changes in this PR looks very confusing to me.

```
<Title as='h1' /> => <h1 href='#' />
```

We also will always call `preventDefault` even for other elements.

```
<TreeTitle as={Link} />
```

Comment (layershifter) [reply packages/react/src/components/Tree/TreeTitle.tsx]: And it actually reverts changes from #1233 as I see.

@priyankar205 please clarify this.

Comment (jurokapsiar) [reply packages/react/src/components/Tree/TreeTitle.tsx]: We had a long discussion couple of months ago with accessibility testers about the structure of the tree. The challenge is that we cannot fully follow the ARIA pattern as it does not work the way how we need for all screen readers. The structure that was agreed upon was implemented and it was changed in #1233 without sufficient testing. @priyankar205 created this PR to revert and I promised him to take a look at the event propagation.

File: packages/react/src/components/Tree/TreeTitle.tsx [hunk 2]
```diff
@@ -57,15 +57,13 @@ class TreeTitle extends UIComponent<ReactProps<TreeTitleProps>> {
   }
 
   public static defaultProps = {
-    as: 'span',
+    as: 'a',
+    href: '#',
```
Comment (jurokapsiar) [inline packages/react/src/components/Tree/TreeTitle.tsx]: @priyankar205 please keep { as: 'a' } but remove href. This is similar to our Menu strucutre. I have also tested with screen readers and it works ok

File: packages/react/src/components/Tree/TreeItem.tsx [hunk 3]
```diff
@@ -103,6 +103,7 @@ class TreeItem extends UIComponent<ReactProps<TreeItemProps>> {
 
   handleTitleOverrides = (predefinedProps: TreeTitleProps) => ({
     onClick: (e, titleProps) => {
+      e.preventDefault()
```
Comment (jurokapsiar) [inline packages/react/src/components/Tree/TreeItem.tsx]: please remove this, preventDefault only in the performClick handler

File: packages/react/src/components/Tree/TreeTitle.tsx (hunk 1 above)
Comment (jurokapsiar) [reply packages/react/src/components/Tree/TreeTitle.tsx]: @priyankar205 preventDefault should be called here, inside of performClick

**Review by jurokapsiar — APPROVED** [review]

**Comment by jurokapsiar** [comment]
changelog not necessary as the information is already provided in the previous push (which has not yet been released)

**Pull request merged**